controllers/usersController.ts

```diff
@@ -1,64 +1,68 @@
-import * as validate from "./utils/validate.js";
 import bcrypt from "bcrypt";
 import { PrismaClient } from "@prisma/client";
 
-import { RequestHandler } from "express";
+import { RequestHandler, Request, Response, NextFunction } from "express";
+
+import { createAccessToken } from "./utils/auth.js";
 
 const prisma = new PrismaClient();
 
 export const getAllUsers: RequestHandler = (req, res) => {
     res.send("hit get users route");
 };
 
-export const registerUser: RequestHandler = async (req, res) => {
+export const registerUser: RequestHandler = async (
+    req: Request,
+    res: Response
+) => {
     const { email, username, firstName, lastName, password } = req.body;
 
-    const newUser = await prisma.user
-        .create({
-            data: {
-                email,
-                username,
-                first_name: firstName,
-                last_name: lastName,
-                password: bcrypt.hashSync(password, 10),
-            },
-        })
-        .then((createdUser: any) => {
-            return createdUser;
-        });
+    const newUser = await prisma.user.create({
+        data: {
+            email,
+            username,
+            first_name: firstName,
+            last_name: lastName,
+            password: bcrypt.hashSync(password, 10),
+        },
+    });
 
     if (newUser) {
         return res.json({
             heading: "Registration successful",
             message:
                 "You can now log in using your registered email or username.",
         });
-    } else {
-        return res.status(500).send({
-            heading: "Server error",
-            message: "Please try again.",
-        });
     }
+
+    return res.status(500).send({
+        heading: "Server error",
+        message: "Please try again.",
+    });
 };
 
-export const checkDuplicateEmail: RequestHandler = async (req, res, next) => {
+export const checkDuplicateEmail: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+) => {
     const { email, username } = req.body;
 
-    const resultEmail = await prisma.user.findUnique({
+    const resultEmail: object | null = await prisma.user.findUnique({
         where: { email: email },
         select: {
             email: true,
         },
     });
 
-    const resultUsername = await prisma.user.findUnique({
+    const resultUsername: object | null = await prisma.user.findUnique({
```

Comment:
can the type be User | null?

Reply:
Yes that's actually what I was going for. Wil read up on the docs and use User instead of object

Reply:
For this one, I ended up using a simple type I defined like {email: string} and {username: string}. 

Another path I could have taken would have been to return the value of the email or username property of the return object and then I can just change the return type to string | null and modify the if checkers after that.

Reply:
For this one I used a simple type like {email: string} or {username: string} since I was only return objects with those fields/properties.

Another option I could have taken would be to add a then to the prisma query then return the value of the email/username property then I can just declare the type of the return to be string | null then change the if checkers below.